### PR TITLE
Location provider cached location

### PR DESCRIPTION
**Pull request by david-rhodes**
Replaces #389, #390

### CODE REVIEW

File: sdkproject/Assets/Mapbox/Unity/Location/EditorLocationProvider.cs [hunk 1]
```diff
@@ -21,28 +22,55 @@ public class EditorLocationProvider : AbstractEditorLocationProvider
...
 		{
 			get
 			{
+				if (_mapInitialized)
+				{
+					var startingLatLong = Conversions.StringToLatLon(_latitudeLongitude);
+					var position = Conversions.GeoToWorldPosition(startingLatLong,
+																 _map.CenterMercator,
+																 _map.WorldRelativeScale).ToVector3xz();
+					position += _targetTransform.position;
```
Comment (atripathi-mb) [inline sdkproject/Assets/Mapbox/Unity/Location/EditorLocationProvider.cs]: Add a null value check for `_transformTransform` , will be nice to be able to use EditorLocationProvider without providing a transform.

File: sdkproject/Assets/Mapbox/Unity/Location/EditorLocationProvider.cs [hunk 2]
```diff
@@ -21,28 +22,55 @@ public class EditorLocationProvider : AbstractEditorLocationProvider
...
+				}
+
 				return Conversions.StringToLatLon(_latitudeLongitude);
 			}
 		}
 
 		protected override void SetLocation()
 		{
-			_currentLocation.Heading = _heading;
+			_currentLocation.Heading = _targetTransform.eulerAngles.y;
```
Comment (atripathi-mb) [inline sdkproject/Assets/Mapbox/Unity/Location/EditorLocationProvider.cs]: Add a null value check for `_transformTransform` , will be nice to be able to use EditorLocationProvider without providing a transform.

Comment (david-rhodes) [reply sdkproject/Assets/Mapbox/Unity/Location/EditorLocationProvider.cs]: How about assigning itself if null, instead?

**Comment by david-rhodes** [comment]
> With the addition of target transform to EditorLocationProvider , do we still need TransformLocationProvider?

Great question! I considered this myself and the only thing holding me back from removing it is that some developer may be using it extensively in their project?

File: sdkproject/Assets/Mapbox/Unity/Location/EditorLocationProvider.cs (hunk 2 above)
Comment (atripathi-mb) [reply sdkproject/Assets/Mapbox/Unity/Location/EditorLocationProvider.cs]: That works too. I'll make that change.

**Comment by atripathi-mb** [comment]
>Great question! I considered this myself and the only thing holding me back from removing it is that some developer may be using it extensively in their project?

Valid point, let's leave it in for now.

**Comment by atripathi-mb** [comment]
** Sample script to test the conversions**
```
using UnityEngine;
using Mapbox.Utils;
using Mapbox.Unity.Utilities;
using Mapbox.Unity.Map;


public class TestConversions : MonoBehaviour
{

	[SerializeField]
	string _latitudeLongitudeString;

	[SerializeField]
	AbstractMap _map;

	// Use this for initialization
	void Start()
	{


	}

	// Update is called once per frame
	void Update()
	{

		if (Input.GetMouseButtonUp(0))
		{
			Vector2d latLong = Conversions.StringToLatLon(_latitudeLongitudeString);
			var geoToWorld = _map.GeoToWorldPosition(latLong);
			Debug.Log("World : " + geoToWorld.ToString());
			var worldToGeo = _map.WorldToGeoPosition(geoToWorld);
			Debug.Log("Geo : " + worldToGeo.ToString());

		}
	}
}
```

**Pull request merged**